x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/lib/is_rate.ts

```diff
@@ -9,21 +9,30 @@ import { has } from 'lodash';
 import { MetricsUIAggregation } from '../../../../../common/inventory_models/types';
 import { SnapshotCustomMetricInput } from '../../../../../common/http_api';
 
-export const isMetricRate = (metric: MetricsUIAggregation): boolean => {
-  const values = Object.values(metric);
-  return values.some((agg) => has(agg, 'derivative')) && values.some((agg) => has(agg, 'max'));
+export const isMetricRate = (metric: MetricsUIAggregation | undefined): boolean => {
+  if (metric != null) {
+    const values = Object.values(metric);
+    return values.some((agg) => has(agg, 'derivative')) && values.some((agg) => has(agg, 'max'));
+  }
+  return false;
 };
 
 export const isCustomMetricRate = (customMetric: SnapshotCustomMetricInput) => {
   return customMetric.aggregation === 'rate';
 };
 
-export const isInterfaceRateAgg = (metric: MetricsUIAggregation) => {
-  const values = Object.values(metric);
-  return values.some((agg) => has(agg, 'terms')) && values.some((agg) => has(agg, 'sum_bucket'));
+export const isInterfaceRateAgg = (metric: MetricsUIAggregation | undefined) => {
```

Comment:
Could we merge `isMetricRate` and `isInterfaceRateAgg` as they are very similar?

My goal is to avoid having the same issue when adding a new function like these and forgetting to check `metric`.

Reply:
These both are used elsewhere in the code, the are distinct enough that we need two seperate checks. I did refactor the code to use type guards from IO-TS.